### PR TITLE
add "Join Event" button for online events

### PR DESCRIPTION
**Pull request by s-cheng**
In this PR, I added the `joinEventLink` field to posts/events. This way we can have a clear button on the event page for people to join online events.

This is based on feedback from EA Anywhere, saying that often people will be confused and not know how to join the event, even if the meeting link is in the event description.

Before the event starts, the button is disabled:
Screen Shot 2022-01-12 at 3 15 04 PM

When the event is soon or now, the button is enabled:
Screen Shot 2022-01-11 at 11 09 06 AM

Matching Facebook's UI, after the event ends I disable the button:
Screen Shot 2022-01-11 at 11 11 09 AM

On mobile, the button moves down so that it doesn't squish the event times:
Screen Shot 2022-01-11 at 11 09 39 AM

Also updated event-related emails to include the link at the top instead of the phrase "Online event":
Screen Shot 2022-01-11 at 11 41 54 AM

----
Outdated screenshot from original PR:
Screen Shot 2022-01-11 at 2 21 14 AM

### CODE REVIEW

**Comment by jpaddison3** [comment]
Maybe the button will be confusing when the event is >3 hours away?

**Comment by s-cheng** [comment]
@jpaddison3 Hmm yeah, I do want to make sure people can easily find the link to join an event anytime before it starts, but maybe the button is confusing. I personally don't like when a site makes me go to their event page right when the event starts in order to see the link to join the event. (Actually I should probably add the join link to the "add to calendar" info as well - I'll add that.)

Maybe the right thing to do is, instead of showing the outlined "Join Event" button, display the link as a URL where "Online event" is currently written.

File: packages/lesswrong/components/posts/AddToCalendar/AddToCalendarIcon.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ const AddToCalendarIcon = ({post, label, hideTooltip, classes}: {
         event={{
           name: post.title,
           details: eventDetails,
-          location: post.location,
+          location: post.onlineEvent ? post.joinEventLink : post.location,
```
Comment (s-cheng) [inline packages/lesswrong/components/posts/AddToCalendar/AddToCalendarIcon.tsx:60]: In my commit just now, I slightly updated the event time format on the event details page and put the `joinEventLink` in the location field of "add to calendar".

Also, now we display the `joinEventLink` URL (truncated) on the event details page under the event time, and the "Join Event" button is disabled until 10 min before the event starts.

Screen Shot 2022-01-12 at 3 15 04 PM

File: packages/lesswrong/lib/collections/posts/custom_fields.ts [hunk 2]
```diff
@@ -720,6 +720,20 @@ addFieldsDict(Posts, {
     type: Date,
     viewableBy: ['guests'],
   },
+  
+  joinEventLink: {
+    type: String,
+    hidden: (props) => !props.eventForm,
+    viewableBy: ['guests'],
+    insertableBy: ['members'],
+    editableBy: ['members', 'sunshineRegiment', 'admins'],
```
Comment (jcmustin) [inline packages/lesswrong/lib/collections/posts/custom_fields.ts:729]: I think the second and third permissions here are redundant, unless I'm missing something. I tried the setup below, and I could still edit the link as an admin:
```suggestion
    editableBy: ['members'],
```

File: packages/lesswrong/components/posts/PostsPage/PostsPageEventData.tsx [hunk 3]
```diff
@@ -1,28 +1,84 @@
...
+    locationNode = joinEventLink ? <a
+      className={classes.onlineEventLocation}
+      href={joinEventLink}
+      title={joinEventLink}
+      target="_blank" rel="noopener noreferrer">
+        {joinEventLink}
+    </a> : <div>Online Event</div>
+  }
+  
+  // before the event starts, the "Join Event" button is disabled
```
Comment (jcmustin) [inline packages/lesswrong/components/posts/PostsPage/PostsPageEventData.tsx:58]: appreciate these comments 🙂

**Review by jcmustin — APPROVED** [review]
Looks great!

**Pull request merged**